src/clj/puppetlabs/services/jruby/jruby_puppet_internal.clj

```diff
@@ -110,6 +139,21 @@
         (.put puppet-config dir (fs/absolute-path value))))
     puppet-config))
 
+(schema/defn borrow-with-timeout-fn :- jruby-schemas/JRubyPuppetBorrowResult
```

Comment:
You didn't do this, but I'm wondering why this is named as `...-fn`.  Does it return a function?